docs/design/SecretsEncryption.md

```diff
@@ -0,0 +1,122 @@
+# Encryption Master Keys
+
+## OVERVIEW
+Previously, all sensitive data (cloud pool credentials, chunk cipher keys, account sync credentials, etc.) was stored plain and un-encrypted in the DB. This poses a security threat in which accessing the PVC of mongo would allow access to all the secrets and through that the data on the system and access to the cloud services.
```

Comment:
```suggestion
Previously, all sensitive data (cloud pool credentials, chunk cipher keys, account sync credentials, etc.) was stored plain and un-encrypted in the DB. This poses a security threat, in which accessing the PVC of mongo would allow access to all the secrets and through that the data on the system and access to the cloud services.
```